action.yml

```diff
@@ -59,52 +66,32 @@ runs:
       if: ${{ github.event_name == 'workflow_dispatch' && inputs.with_ssh_enabled }}
       with:
         detached: true
-
-    - name: Install Terraform and Terragrunt
-      shell: bash
-      env:
-        TF_ENV_VERSION: v3.0.0
-        TG_SWITCH_VERSION: 0.6.0
-      run: |
-        rm -rf ~/.tfenv
-        rm -rf "${HOME}/tgswitch"
-        # Install tfenv
-        git clone --depth=1 --branch ${TF_ENV_VERSION} https://github.com/tfutils/tfenv.git ~/.tfenv
-
-        # Install tgswitch
-        mkdir -p "${HOME}/tgswitch"
-        wget -q https://github.com/warrensbox/tgswitch/releases/download/${TG_SWITCH_VERSION}/tgswitch_${TG_SWITCH_VERSION}_linux_amd64.tar.gz -O /tmp/tgswitch_${TG_SWITCH_VERSION}_linux_amd64.tar.gz
-        tar -xzf /tmp/tgswitch_${TG_SWITCH_VERSION}_linux_amd64.tar.gz -C ${HOME}/tgswitch
-        chmod u+x ${HOME}/tgswitch/tgswitch
-        rm -rf /tmp/tgswitch_${TG_SWITCH_VERSION}_linux_amd64.tar.gz
-
-        # Updating PATH for future steps
-        echo "$HOME/.tfenv/bin" >> "$GITHUB_PATH"
-        echo "$HOME/tgswitch" >> "$GITHUB_PATH"
-
-    - name: Install Terraform and Terragrunt Versions
+    - name: Setup Mise Toml
+      id: mise-toml
       shell: bash
       env:
         TF_VERSION: ${{ inputs.tf_version }}
         TG_VERSION: ${{ inputs.tg_version }}
+        TF_BINARY: ${{ inputs.tf_binary }}
       run: |
-        # Install Terraform and Terragrunt
-        tfenv install "$TF_VERSION"
-        tfenv use "$TF_VERSION"
-        tgswitch "$TG_VERSION"
+        echo 'TOML<<EOF' >> "$GITHUB_OUTPUT"
+        echo '[tools]' >> "$GITHUB_OUTPUT"
+        echo "$TF_BINARY = \"$TF_VERSION\"" >> "$GITHUB_OUTPUT"
+        echo "terragrunt = \"$TG_VERSION\"" >> "$GITHUB_OUTPUT"
+        echo 'EOF' >> "$GITHUB_OUTPUT"
+    - uses: jdx/mise-action@v2
+      with:
+          install: true
+          cache: true
+          mise_toml: "${{ steps.mise-toml.outputs.TOML }}"
 
-    - name: Test Terraform and Terragrunt
+    - name: Test Terraform, OpenTofu and Terragrunt
       shell: bash
       run: |
-        terraform --version
+        terraform --version || true
+        tofu --version || true
```

Comment:
GitHub actions has a pre-installed terraform version so i would expect that a terraform version would also show up in the logs. Can this be confusing? I also could not check the linked pipelines run due to permissions.

Reply:
Indeed, the ubuntu baseimage comes with TF 1.7.4.  Its not awesome that its there regardless, but we do set the envvar manually to tell TG what to use below.